Move role attribute in “<pre>: The Preformatted Text element” (#12878)
mdn/content: files/en-us/web/html/element/pre/index.md

People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.

> A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}

Reviewer: ```suggestion
A combination of {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
```

> and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes, allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.

Reviewer 2: removing the `role=img` makes this line not make sense anymore.

"allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description."

without the `role=img` there is no longer an "image".

Rather this should be revised further to resemble [this example from ARIA in HTML](https://w3c.github.io/html-aria/#example-8).

> A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of an {{htmlattrxref("id")}}
> and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.

Reviewer: ```suggestion
A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by a combination of {{htmlattrxref("id")}}
and [`aria-labelledby`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) attributes, allows the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.
```

People experiencing low vision conditions and browsing with the aid of assistive technology such as a screen reader may not understand what the preformatted text characters are representing when they are read out in sequence.

> A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) attributes on the `pre` element allow the preformatted text to be announced as an image, with the `figcaption` serving as the image's alternate description.

Reviewer 2: ```suggestion
A combination of the {{HTMLElement("figure")}} and {{HTMLElement("figcaption")}} elements, supplemented by the [ARIA](/en-US/docs/Web/Accessibility/ARIA) `role` and [`aria-label`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label) attributes on the `pre` element allow the preformatted ASCII art to be announced as an image with alternative text, and the `figcaption` serving as the image's caption.
```
